File: articles/hypocrisy-dehumanization-and-the-psychology-of-moral-failure/c/moral_failure_probability.c
/*
Synthetic moral failure probability example.
Educational scaffold only.

Compile:
  cc c/moral_failure_probability.c -lm -o outputs/tables/moral_failure_probability_c
Run:
  ./outputs/tables/moral_failure_probability_c
*/

#include <math.h>
#include <stdio.h>

double logistic(double x) {
    return 1.0 / (1.0 + exp(-x));
}

int main(void) {
    double hypocrisy = 0.8;
    double dehumanization = 0.7;
    double ingroup_bias = 0.5;
    double institutional_hypocrisy = 0.6;
    double empathic_recognition = 0.3;
    double accountability_strength = 0.4;
    double victim_visibility = 0.2;

    double latent =
        0.45 * hypocrisy +
        0.40 * dehumanization +
        0.30 * ingroup_bias +
        0.25 * institutional_hypocrisy -
        0.35 * empathic_recognition -
        0.25 * accountability_strength -
        0.20 * victim_visibility;

    double probability = logistic(latent);

    printf("Synthetic moral failure probability: %.3f\n", probability);
    return 0;
}

File: articles/hypocrisy-dehumanization-and-the-psychology-of-moral-failure/rust/moral_failure_probability.rs
// Synthetic moral failure probability example.
// Educational scaffold only.
//
// Compile:
//   rustc rust/moral_failure_probability.rs -o outputs/tables/moral_failure_probability_rust
// Run:
//   ./outputs/tables/moral_failure_probability_rust

fn logistic(x: f64) -> f64 {
    1.0 / (1.0 + (-x).exp())
}

fn main() {
    let hypocrisy = 0.8_f64;
    let dehumanization = 0.7_f64;
    let ingroup_bias = 0.5_f64;
    let institutional_hypocrisy = 0.6_f64;
    let empathic_recognition = 0.3_f64;
    let accountability_strength = 0.4_f64;
    let victim_visibility = 0.2_f64;

    let latent =
        0.45 * hypocrisy
        + 0.40 * dehumanization
        + 0.30 * ingroup_bias
        + 0.25 * institutional_hypocrisy
        - 0.35 * empathic_recognition
        - 0.25 * accountability_strength
        - 0.20 * victim_visibility;

    let probability = logistic(latent);

    println!("Synthetic moral failure probability: {:.3}", probability);
}
